controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/OpsDbInfoUpdater.java

```diff
@@ -73,8 +74,8 @@ private static String registeredHostnameOf(Node host) {
 
     private static boolean shouldUpdate(Node node, NodeEntity nodeEntity) {
         if (nodeEntity == null) return false;
-        if (nodeEntity.switchHostname().isEmpty()) return false;
-        return !node.switchHostname().equals(nodeEntity.switchHostname());
+        if (nodeEntity.switchHostname().isEmpty() || nodeEntity.model().isEmpty())  return false;
```

Comment:
This looks wrong to me. If switch is non-empty, but model is empty we still may want to update the switch name and vice versa. 

You can create two separate methods: `shouldUpdateSwitch` and `shouldUpdateModel`. Patching can be skipped if both are `false`.

Reply:
👍